RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,111 @@
+# Release Checklist
+
+This is a list of the things that need to happen during a release.
+
+## Build a Release
+
+### Prepare the Changelog (Full release only)
```

Comment:
idk if we do release candidates for packages; i kinda think no? so we can remove copy distinguishing between RCs and full releases i think